ext/php8/php8/engine_hooks.c

```diff
@@ -738,6 +738,11 @@ static void dd_observer_end_handler(zend_execute_data *execute_data, zval *retva
 
 zend_observer_fcall_handlers ddtrace_observer_fcall_init(zend_execute_data *execute_data) {
     zend_function *fbc = EX(func);
+
+    if (!PG(modules_activated)) {
```

Comment:
To call out a behavior difference on PHP 8: any extension that runs code in RINIT would essentially disable the tracer's ability to trace any function call that is called in RINIT due to installing NULL observer handlers. This is certainly an edge case but could cause some unexpected behavior that might be a bit tricky to debug.

I wonder if emitting a log message for these cases would help with debugging? It might have to be a one-time log message so that it is not so noisy.